src/twisted/internet/endpoints.py

```diff
@@ -888,6 +1016,22 @@ def cancelRemainingPending(result):
         return d
 
 
+    def _nameResolution(self, host, port):
+        """
+        Resolve the hostname string into a tuple containing the host
+        address.
+
+        @param host: A unicode hostname to resolve.
+
+        @param port: The port to include in the resolution.
+
+        @return: A L{Deferred} that fires with L{_getaddrinfo}'s
+            return value.
+        """
+        return self._deferToThread(self._getaddrinfo, host, port, 0,
+                socket.SOCK_STREAM)
```

Comment:
`deferToThread` has to be replaceable to make this function testable.  The original implementation also made `getaddrinfo` replaceable, apparently with the intent of verifying that `deferToThread` invokes it.

I was inclined to leave these two staticmethods in for two reasons:

1. My tests looked more like the existing ones, apparently verifying the same behaviors.
2.  A vague concern that some tests, inside or outside Twisted itself, relied on it.  

A cursory GitHub search indicated that 2. wasn't the case.  1. remains true and useful, I think.